r/vignettes/developing.Rmd

```diff
@@ -60,36 +71,38 @@ brew install apache-arrow
 brew install apache-arrow --HEAD
 ```
 
+### Windows and Linux
+
 On Windows and Linux, you can download a .zip file with the arrow dependencies from the
 nightly repository.
-Windows users then can set the `RWINLIB_LOCAL` environment variable to point to that
-zip file before installing the `arrow` R package. On Linux, you'll need to create a `libarrow` directory inside the R package directory and unzip that file into it. Version numbers in that
-repository correspond to dates, and you will likely want the most recent.
 
 To see what nightlies are available, you can use Arrow's (or any other S3 client's) S3 listing functionality to see what is in the bucket `s3://arrow-r-nightly/libarrow/bin`:
 
 ```
 nightly <- s3_bucket("arrow-r-nightly")
 nightly$ls("libarrow/bin")
 ```
+Version numbers in that repository correspond to dates.
 
-## Developer environment setup
+#### Windows
 
-If you need to alter both the Arrow C++ library and the R package code, or if you can’t get a binary version of the latest C++ library elsewhere, you’ll need to build it from source too. This section discusses how to set up a C++ build configured to work with the R package. For more general resources, see the [Arrow C++ developer
-guide](https://arrow.apache.org/docs/developers/cpp/building.html).
+Windows users then can set the `RWINLIB_LOCAL` environment variable to point to the zip file containing the arrow dependencies before installing the arrow R package. 
+
+#### Linux
+
+On Linux, you'll need to create a `libarrow` directory inside the R package directory and unzip the zip file containing the arrow dependencies into it. 
+
+## R and C++
 
-There are four major steps to the process — the first three are relevant to all Arrow developers, and the last one is specific to the R bindings:
+If you need to alter both the Arrow C++ library and the R package code, or if you can't get a binary version of the latest C++ library elsewhere, you'll need to build it from source. This section discusses how to set up a C++ build configured to work with the R package. For more general resources, see the [Arrow C++ developer guide](https://arrow.apache.org/docs/developers/cpp/building.html).
 
-1. Configuring the Arrow library build (using `cmake`) — this specifies how you want the build to go, what features to include, etc.
-2. Building the Arrow library — this actually compiles the Arrow library
-3. Install the Arrow library — this organizes and moves the compiled Arrow library files into the location specified in the configuration
-4. Building the R package — this builds the C++ code in the R package, and installs the R package for you
+There are five major steps to the process — the first four are relevant to all Arrow developers, and the last one is specific to developers making changes to the R package.
```

Comment:
This is a very minor point, so we might want to ignore it totally / gloss over it here, but technically "the first four are relevant to all Arrow developers" isn't _quite_ true. There are (small) differences in building the cpp code between the languages that use it (for example the flags recommended [for python development](https://arrow.apache.org/docs/developers/python.html#build-and-test) are slightly different — we could in principle recommend that folks use a union of the flags and then they would be able to do either, though compilation would be longer, so I'm not sure we want to do that). Additionally, other languages, like Rust, don't rely on the cpp code at all AFAIK, so for _those_ arrow developers this isn't super relevant.

All of that said, it might be best to leave it as is, or say something like "are similar to steps that Arrow developers working on other languages use" and leave it be. The details above in this comment are definitely *not* relevant for someone building Arrow for R development for the first time

Reply:
Thanks for highlighting this, I'll come up with a different way of phrasing this to be more accurate!

Reply:
I've just deleted it as it doesn't really add anything.